### ISSUE TITLE
PHP 8.3 - Typed class constants support

### ISSUE THREAD

**Issue by jnoordsij** [squizlabs/PHP_CodeSniffer#3927]
## Describe the bug
PHP 8.3 allows defining a type for class constants (see https://php.watch/versions/8.3/typed-constants) to enforce compatibility with overrides and have better static analysis support.

When using this with current stable `phpcs`, this triggers an error message like:
```
Class constants must be uppercase; expected STRING but found string
```

### Code sample
```php
class Test {
    const string TEST_CONSTANT = 'test';
}
```

### Custom ruleset
```xml
<?xml version="1.0"?>
<ruleset name="My Custom Standard">
  <description>If you are using a custom ruleset, please enter it here.</description>
</ruleset>
```

### To reproduce
1. Create a file called `test.php` with the code sample above...
2. Run `phpcs test.php ...`
3. See error message displayed
```
Class constants must be uppercase; expected STRING but found string
```

## Expected behavior
No error.

## Versions (please complete the following information)

| | |
|-|-|
| Operating System | any
| PHP version | 8.3
| PHP_CodeSniffer version | latest
| Standard | any?
| Install type | composer

## Please confirm:

- [x] I have searched the issue list and am not opening a duplicate issue.
- [x] I confirm that this bug is a bug in PHP_CodeSniffer and not in one of the external standards.
- [x] I have verified the issue still exists in the `master` branch of PHP_CodeSniffer.

**Comment by MaxiCom** [comment]
I encountered the same bug today, looking forward a fix! :wave:

**Comment by matmper** [comment]
Same problem, for now I'm ignoring my Enum folder.

**Comment by PatNowak** [comment]
Any news on that?

**Comment by ldebrouwer** [comment]
@PatNowak Please take a moment to read; https://github.com/squizlabs/PHP_CodeSniffer/issues/3932

**Comment by jrfnl** [comment]
FYI: a fix for this issue is included in today's [PHP_CodeSniffer 3.9.0 release](https://github.com/PHPCSStandards/PHP_CodeSniffer/releases/tag/3.9.0).

As per #3932, development on PHP_CodeSniffer will continue in the [PHPCSStandards/PHP_CodeSniffer](https://github.com/PHPCSStandards/PHP_CodeSniffer) repository. If you want to stay informed, you may want to start "watching" that repo (or watching releases from that repo).
